prototype-spec.md

```diff
@@ -0,0 +1,253 @@
+* `KEYID` is the unique identifier of the key signing the `ROLE` object.
+* `SIGNATURE` is a hex-encoded signature of the canonical form of the metadata for `ROLE`.
+
+### Keys
+
+Keys will be described in the following format:
+
+```
+{
+  "keytype" : KEYTYPE,
+  "scheme" : SCHEME,
+  "keyval" : KEYVAL
+}
+```
+
+where
+
+* `KEYTYPE` is a string denoting a public key signature system, such as `rsa`, `ed25519` or `ecdsa-sha2-nistp256`.
+* `SCHEME` is a string denoting a corresponding signature scheme.  For example: `rsassa-pss-sha256`, `ed25519`, and `ecdsa-sha2-nistp256`.
+* `KEYVAL` is a dictionary containing the public portion of the key.
+
+### Root
+
+Root metadata will contain the following in the `ROLE` field:
+
+```
+{
+  "_type" : "root",
+  "spec_version" : SPEC_VERSION,
+  "consistent_snapshot": CONSISTENT_SNAPSHOT,
+  "version" : VERSION,
+  "expires" : EXPIRES,
+  "keys" : {
+    KEYID : KEY,
+    ...
+  },
+  "roles" : {
+    ROLE : {
+      "keyids" : [
+        KEYID,
+        ...
+      ] ,
+      "threshold" : THRESHOLD
+    },
+    ...
+  }
+}
+```
+
+where
+* `SPEC_VERSION` is a string that contains the version number of the TUF specification. Its format follows the Semantic Versioning 2.0.0 (semver) specification. Metadata is written according to version "spec_version" of the specification, and clients MUST verify that "spec_version" matches the expected version number. Adopters are free to determine what is considered a match (e.g., the version number exactly, or perhaps only the major version number (major.minor.fix).
+* `CONSISTENT_SNAPSHOT` is a boolean indicating whether the repository supports consistent snapshots. Section 7 goes into more detail on the consequences of enabling this setting on a repository.
+* `VERSION` is an integer that is greater than 0. Clients MUST NOT replace a metadata file with a version number less than the one currently trusted.
+* `EXPIRES` is a date-time string indicating when metadata should be considered expired and no longer trusted by clients. Clients MUST NOT trust an expired file.
+* `ROLE` is one of "root", "snapshot", "targets", or "timestamp". A role for each of "root", "snapshot", "timestamp", and "targets" MUST be specified in the key list.
+* `KEYID` is a KEYID, which MUST be correct for the specified KEY. Clients MUST ensure that for any KEYID represented in this key list and in other files, only one unique key has that KEYID.
+* `THRESHOLD` is an integer number of keys of that role whose signatures are required in order to consider a file as being properly signed by that role.
+
+### Snapshot
+
+Snapshot metadata will contain the following in the `ROLE` field:
+
+```
+{
+  "_type" : "snapshot",
+  "spec_version" : SPEC_VERSION,
+  "version" : VERSION,
+  "expires" : EXPIRES,
+  "meta" : METAFILES
+}
+```
+
+* `SPEC_VERSION`, `VERSION` and `EXPIRES` are the same as is described for the root.json file.
+* METAFILES is an object whose format is the following:
```

Comment:
quibbling here, but "metafiles" is plural, so can you say is an object? Shouldn't it be "are objects?"